### PR TITLE
Allow deploy.rb to build against a PR 

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

**Comment by ehelms** [comment]
What's the scenario for this change?

**Comment by stbenjam** [comment]
If you submit a PR against KATELLO-2.1, I create a git repo with your PR merged against KATELLO-2.1, and then I need deploy.rb to use that -- not katello.org.

**Comment by stbenjam** [comment]
See: https://github.com/Katello/katello.org/pull/120/files#diff-52c976fc38ed2b4e3b1192f8a8e24cffR21

**Comment by stbenjam** [comment]
But I'm almost certain this won't do what I want *sigh* I misunderstood deploy.rb, I thought it would use itself (its own git checkout) as the base for deploying.

I'll have to look tomorrow :-(

**Comment by stbenjam** [comment]
New version here works much better.  `./deploy.rb --pr 120` will compile the web site with the changes from a particular PR.

And then the Rake task in the other PR runs the tests against it.

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+source "http://rubygems.org"
+
+group :development do
+  gem 'therubyracer'
+  gem 'jekyll'
+  gem 'kramdown'
+  gem 'pygments.rb'
+  gem 'rake'
+  gem 'html-proofer'
+end
```
Comment (ehelms) [inline Gemfile]: Why is this needed at the top level in the branch?

Comment (stbenjam) [inline Gemfile]: Travis needs the Gemfile to execute deploy.rb in the context of bundler, otherwise it fails.  Why wouldn't it make sense to have a Gemfile here if deploy.rb relies on gems?

Comment (ehelms) [inline Gemfile]: I don't see the script relying on the above gems?

Comment (stbenjam) [inline Gemfile]: Jekyll? https://github.com/Katello/katello.org/blob/deploy/deploy.rb#L127

I can probably get by without rake and html-proofer.

Comment (ehelms) [inline Gemfile]: I guess that method could be updated -- if you see where its called from, it's always from within the build directory that the bundle install just above it is called from. My goal was to allow master to be the, well, master of the Gemfile.

File: deploy.rb [hunk 2]
```diff
@@ -64,6 +76,7 @@ def build_nightly
     reset_nightly
 
     Dir.chdir(BUILD_DIR + '/nightly') do
+      build_pr_branch('master')
```
Comment (ehelms) [inline deploy.rb]: Minor but maybe move the `Building Nightly` message above below this line so that the output would read:

```
Testing PR 77 against branch master
Building nightly...
Building version 2.0
Copying 2.0
Building version 2.1
Copying 2.1
```

File: Gemfile (hunk 1 above)
Comment (stbenjam) [inline Gemfile]: Ah ok, that makes sense...

I'll try to figure out what travis is doing then that bundler is complaining, maybe there's nothing wrong here.

Comment (ehelms) [inline Gemfile]: Its possible it doesn't like that we are bundling from another directory or that it is trying to bundle by default for a ruby project?

File: deploy.rb (hunk 2 above)
Comment (stbenjam) [inline deploy.rb]: Sure, looks better. Now like this:

```
Testing PR 120 against branch master
Detecting available Katello versions
Versions found: ["2.0", "2.1"]
Building nightly...
Building version 2.0
Copying 2.0
Building version 2.1
Copying 2.1
```

**Comment by ehelms** [comment]
ACK
